### PR TITLE
[PFS-151][PFS-152][PFS-157] New squash / drop commit RPCs and implementations

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR adds new squash / drop commit RPCs with the semantics proposed here: https://www.notion.so/2023-08-21-Squash-Commit-851a24fd5df0480c82b05f643048a48d.

### CODE REVIEW

File: src/pfs/pfs.proto [hunk 1]
```diff
@@ -305,6 +305,20 @@ message ClearCommitRequest {
   Commit commit = 1;
 }
 
+message SquashCommitRequest {
+  Commit commit = 1;
+  bool recursive = 2;
```
Comment (bbonenfant) [inline src/pfs/pfs.proto:310]: the commit field is self explanatory, but could you add a comment here (and for the DropCommitRequest message) for the recursive field.

**Review by bbonenfant — APPROVED** [review]

**Review by brendoncarroll — APPROVED** [review]
Agree with @bbonenfant about comments in the protobufs

File: src/server/pfs/cmds/cmds.go [hunk 2]
```diff
@@ -895,6 +895,32 @@ func Cmds(mainCtx context.Context, pachCtx *config.Context, pachctlCfg *pachctl.
...
+			_, err = c.PfsAPIClient.SquashCommit(c.Ctx(), &pfs.SquashCommitRequest{
+				Commit:    commit,
+				Recursive: recursive,
+			})
+			return err
+		}),
+	}
+	shell.RegisterCompletionFunc(squashCommitV2, shell.BranchCompletion)
+	squashCommitV2.Flags().BoolVarP(&recursive, "recursive", "r", false, "Recursively squash all subvenant commits")
+	commands = append(commands, cmdutil.CreateAliases(squashCommitV2, "squash commitV2"))
```
Comment (acohen4) [inline src/server/pfs/cmds/cmds.go:922]: should we just name this `squash`?

**Review by acohen4 — APPROVED** [review]
LGTM

**Pull request merged**